### PR TITLE
Vet clinic: query and update animals table

### PR DESCRIPTION
**Pull request by raihan2bd**
# Vet Clinic 🐻 
Hello

### Hi There 👋 In this branch I made the following changes ⬇️ 
- [ ] I added a `species` column inside the `animals` table ✅ 
01  species

- [ ] Then I `Insert` More data inside the `animals` table ✅ 
02  Insert_more_data_to_the_animals_table

- [ ] Then Rollback species column with `unspecified` ✅ 
03  Species_rollback

- [ ] Then Inside a transaction I update the `animals` table species column by name ✅ 
04  update_species_by_name

- [ ] Inside a transaction I deleted all records in the `animals` table, then roll back the transaction ✅ 
05 delete_animals_data_and_rollback

- [ ] Inside a transaction: ✔️ 
    - Deleted all animals born after `Jan 1st, 2022`.
    - Created a `savepoint` for the transaction.
    - Updated all `animals'` weights to be their weight multiplied by -1.
    - `Rollbacked` to the savepoint
    - Updated all animals' negative weights to be their weight `multiplied by -1`.
    - `Committed` transaction
06  update_animals_table_by_weight_and_birth_date

- [ ] Then added necessary queries to select and count animals' table data according to the project requirment✅ 
07  following_questions_part_1
08  following_questions_part_2

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -23,3 +23,45 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+-- Query to select animals table data with specific condition and also count them
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT ROUND(AVG(weight_kg)::numeric, 2) FROM animals;
+SELECT name FROM animals 
+  WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
+SELECT species, MIN(weight_kg), MAX(weight_kg)
```
Comment (Shedrack-Sunday) [inline queries.sql]: - [x] Based on project requirements, kindly specify your query on Who escapes the most, neutered or not neutered animals. 

## Sample code:
` SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered
`

## Instructions
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @raihan2bd       👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing some of the requirements,  However, there are a few you need to work on, to move to the next project. 💯

 


## Highlights 
- Nice setup of linters✔️  
- Good SQL query format used   ✔️  
  
# Required Changes  🌵 

- [ ] There are a few things you need to take note of to have a great project as expected 👍 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: queries.sql (hunk 1 above)
Comment (raihan2bd) [reply queries.sql]: Hi, @Shedrack-Sunday 👋 Thanks a lot for your valuable suggestions. I will make changes according to your suggestion.

**Review by AlphaNtihinduka — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 
## Hi @raihan2bd 🙋‍♂️, 

Your project is complete!  There is nothing else to say other than... Great job 👏 

approved-accepted

Congratulations! 🎉 
TO HIGHLIGHT:

- Your project is professional :heavy_check_mark: 
- Your repo is professional :heavy_check_mark: 
- Well-structured files :heavy_check_mark: 
- Professional README.md file 💯
- No linter errors :heavy_check_mark: 

Now you can merge this branch and move on :1st_place_medal: 


## Optional suggestions
- Kindly refer to the inline optional suggestions made. Ensure to apply them to have a professional project that suits recruiters' needs.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :four_leaf_clover:

**Pull request merged**